src/Entities/ProjectEntity.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace ProjMange\Entities;
+
+class ProjectEntity implements \JsonSerializable
+{
+    private int $id;
+    private string $name;
+    private int $client_id;
+    private $deadline;
+
+
+    /**
+     * Calculates whether the project is overdue
+     * @return bool
+     */
+    private function calculateOverdue(): bool
+    {
+        return date($this->deadline) <= date('Y-m-d')
```

Comment:
overdue should be from the day after the deadline

Reply:
this should be a service. replace this function with a plain getter to retrieve the deadline and then in your abstract static service method you can accept a date string and evaluate if overdue.

Reply:
plus you can unit test the service (yay)